android-lib/src/androidTest/java/com/kinvey/androidTest/store/DataStoreTest.java

```diff
@@ -1489,6 +1504,190 @@ public void testPagedPullBlocking() throws InterruptedException, IOException {
         assertTrue(pullResults.size() == getCacheSize(StoreType.CACHE));
     }
 
+    @Test
+    public void testSkipLimitInPullBlocking() throws InterruptedException, IOException {
+        DataStore<Person> store = DataStore.collection(Person.COLLECTION, Person.class, StoreType.CACHE, client);
+        client.getSyncManager().clear(Person.COLLECTION);
+
+        cleanBackendDataStore(store);
+
+        // Arrange
+        ArrayList<Person> persons = new ArrayList<>();
+        Person alvin = createPerson("Alvin");
+        Person simon = createPerson("Simon");
+        Person theodore = createPerson("Theodore");
+        Person anna = createPerson("Anna");
+        Person kate = createPerson("Kate");
+        save(store, alvin);
+        save(store, simon);
+        save(store, theodore);
+        save(store, anna);
+        save(store, kate);
+        long cacheSizeBefore = getCacheSize(StoreType.CACHE);
+        assertTrue(cacheSizeBefore == 5);
+        client.getCacheManager().getCache(Person.COLLECTION, Person.class, StoreType.CACHE.ttl).clear();
+        long cacheSizeBetween = getCacheSize(StoreType.CACHE);
+        assertTrue(cacheSizeBetween == 0);
+
+        List<Person> pullResults = null;
+        Query query = client.query();
+        for (int i = 0; i < 5; i++) {
+            query.setLimit(1);
+            query.setSkip(i);
+            pullResults = store.pullBlocking(query).getResult();
+
+            assertNotNull(pullResults);
+            assertTrue(pullResults.size() == 1);
+            assertEquals(i+1, getCacheSize(StoreType.CACHE));
+        }
+        assertEquals(5, getCacheSize(StoreType.CACHE));
+        for (int i = 0; i < 5; i++) {
+            query.setLimit(1);
+            query.setSkip(i);
+            pullResults = store.pullBlocking(query).getResult();
+
+            assertNotNull(pullResults);
+            assertTrue(pullResults.size() == 1);
+            assertEquals(5, getCacheSize(StoreType.CACHE));
+        }
```

Comment:
Would it make sense in these tests to check each entity as it is returned to verify that it is the one that we expect, since we have a sort order?

Reply:
It makes sense because in this way I check that sorting works correct after getting the cache. If sorting won't work correct user can lose them some data.